src/util/sighash.rs

```diff
@@ -0,0 +1,912 @@
+
+        // If hash_type & 0x80 equals SIGHASH_ANYONECANPAY:
+        //      outpoint (36): the COutPoint of this input (32-byte hash + 4-byte little-endian).
+        //      amount (8): value of the previous output spent by this input.
+        //      scriptPubKey (35): scriptPubKey of the previous output spent by this input, serialized as script inside CTxOut. Its size is always 35 bytes.
+        //      nSequence (4): nSequence of this input.
+        if anyone_can_pay {
+            let txin =
+                &self
+                    .tx
+                    .input
+                    .get(input_index)
+                    .ok_or_else(|| Error::IndexOutOfInputsBounds {
+                        index: input_index,
+                        inputs_size: self.tx.input.len(),
+                    })?;
+            let previous_output = prevouts.get(input_index)?;
+            txin.previous_output.consensus_encode(&mut writer).unwrap();
+            previous_output.value.consensus_encode(&mut writer).unwrap();
+            previous_output
+                .script_pubkey
+                .consensus_encode(&mut writer)
+                .unwrap();
+            txin.sequence.consensus_encode(&mut writer).unwrap();
+        } else {
+            (input_index as u32).consensus_encode(&mut writer).unwrap();
+        }
+
+        // If an annex is present (the lowest bit of spend_type is set):
+        //      sha_annex (32): the SHA256 of (compact_size(size of annex) || annex), where annex
+        //      includes the mandatory 0x50 prefix.
+        if let Some(annex) = annex {
+            let mut enc = sha256::Hash::engine();
+            annex.consensus_encode(&mut enc).unwrap();
+            let hash = sha256::Hash::from_engine(enc);
+            hash.consensus_encode(&mut writer).unwrap();
+        }
+
+        // * Data about this output:
+        // If hash_type & 3 equals SIGHASH_SINGLE:
+        //      sha_single_output (32): the SHA256 of the corresponding output in CTxOut format.
+        if sighash == SigHashType::Single {
+            let mut enc = sha256::Hash::engine();
+            self.tx
+                .output
+                .get(input_index)
+                .ok_or_else(|| Error::SingleWithoutCorrespondingOutput {
+                    index: input_index,
+                    outputs_size: self.tx.output.len(),
+                })?
+                .consensus_encode(&mut enc)
+                .unwrap();
+            let hash = sha256::Hash::from_engine(enc);
+            hash.consensus_encode(&mut writer).unwrap();
+        }
+
+        //     if (scriptpath):
+        //         ss += TaggedHash("TapLeaf", bytes([leaf_ver]) + ser_string(script))
+        //         ss += bytes([0])
+        //         ss += struct.pack("<i", codeseparator_pos)
+        if let Some(ScriptPath {
+            script,
+            leaf_version,
+            code_separator_pos,
+        }) = script_path
+        {
```

Comment:
nit, this fits in a single 93-character line which is shorter than the 100 char column limit, no?

Reply:
I just used default rustfmt for this file

Reply:
Oh wow, I wouldn't have complained here, but thought about leaving a nit for L345-353. Sometimes `rustfmt` seems quite off. But I think it's ok if it follows a consistent standard.

Reply:
since it needs re-approval, would you like 844dfe2dfce31e966daff9150ac613b734cc1f97 to improve fmt around line L345-353 ?

Reply:
Nah, we want to use `rustfmt` eventually anyway. If that's its output I'm fine with it.

Reply:
844dfe2 is always following `rustfmt` rules, but the added let binding change its behaviors